packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Statements/InvokeStaticMethodStatement.cs

```diff
@@ -7,13 +7,30 @@
 
 namespace Microsoft.Generator.CSharp.Statements
 {
-    public sealed record InvokeStaticMethodStatement(CSharpType? MethodType, string MethodName, IReadOnlyList<ValueExpression> Arguments, IReadOnlyList<CSharpType>? TypeArguments = null, bool CallAsExtension = false, bool CallAsAsync = false) : MethodBodyStatement
+    public sealed class InvokeStaticMethodStatement : MethodBodyStatement
     {
+        public CSharpType? MethodType { get; }
+        public string MethodName { get; }
+        public IReadOnlyList<ValueExpression> Arguments { get; }
+        public IReadOnlyList<CSharpType>? TypeArguments { get; }
+        public bool CallAsExtension { get; }
+        public bool CallAsAsync { get; }
+
+        public InvokeStaticMethodStatement(CSharpType? methodType, string methodName, IReadOnlyList<ValueExpression> arguments, IReadOnlyList<CSharpType>? typeArguments = null, bool callAsExtension = false, bool callAsAsync = false)
+        {
+            MethodType = methodType;
+            MethodName = methodName;
+            Arguments = arguments;
```

Comment:
and we could just have
```suggestion
            Arguments = arguments ?? Array.Empty<CSharpType>();
```